CONTRIBUTING.md

```diff
@@ -0,0 +1,185 @@
+
+If the tests ran successfully, there are usually two possible outcomes:
+a) If the bug has not appeared and the tests have passed successfully, ask the reporter to provide more details. The bug can be not a bug actually or can relate to a particular software version used or specifics of local environment configuration.
+
+b) The bug has appeared and the tests has failed as expected showing the reported symptoms.
+
+14. Fix the bug.
+
+15. Run `flake8` against a changed file. If it is `plugins/modules/mysql_user.py`:
+```bash
+flake8 plugins/modules/mysql_user.py
+```
+It is worth installing and running `flake8` against the changed file(s) first.
+It shows unused imports, which is not shown by sanity tests (see the next step), as well as other common issues.
+Optionally, you can use the `--max-line-length=160` command-line argument.
+
+16. Run sanity tests:
+```bash
+ansible-test sanity plugins/modules/mysql_user.py --docker
+```
+If they failed, look at the output carefully - it is usually very informative and helps to identify a problem line quickly.
+Sanity failings usually relate to wrong code and documentation formatting.
+
+17. Run integration tests:
+```bash
+ansible-test integration test_mysql_user --docker -vvv
+```
+
+There are two possible outcomes:
+a) They have failed. Look at the output of the command.
+Fix the problem place in the code and run again.
+Repeat the cycle until the tests pass.
+
+b) They have passed. Remember they failed originally? Our congratulations! You has probably fixed the bug, though we hope not introducing a couple of new ones;)
+
+18. Commit your changes with an informative but short commit message:
+```bash
+git add /path/to/changed/file
+git commit -m "mysql_user: fix crash when ..."
+```
+
+19. Push the branch to the `origin` (your fork):
+```bash
+git push origin name_of_my_branch
+```
+
+20. Go to the `upstream` (http://github.com/ansible-collections/community.mysql).
+
+21. Go to `Pull requests` tab and create a pull request.
+
+GitHub is tracking your fork, so it should see the new branch in it and automatically offer
+to create a pull request. Sometimes GitHub does not do it and you should click the `New pull request` button yourself.
+Then choose `compare across forks` under the `Compare changes` title.
+Choose your repository and the new branch you pushed in the right drop-down list.
+Confirm. Fill out the pull request template with all information you want to mention.
+Put "Fixes + link to the issue" in the pull request's description.
+Put "[WIP] + short description" in the pull request's title. It's often a good idea to mention the name of the module/plugin you are modifying at the beginning of the description.
+Click `Create pull request`.
+
+22. Add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs) to the `changelog/fragments` directory. It will be published in release notes, so users will know about the fix.
```

Comment:
@Andersson007 This should be step 18, or the changelog fragment will not make it to the PR

Reply:
@Jorge-Rodriguez the problem is that we often don't know a PR number until it's created (it's technically possible to predict but it would be hard to explain how).

Reply:
In that case, we're missing step 23 (and 24?) to commit and push the changelog fragments

Reply:
@Jorge-Rodriguez very good spotted, thanks! I added a comment and bash code fragment to step 22